features/step_definitions/advocate_claim_steps.rb

```diff
@@ -1,6 +1,9 @@
-
 include WaitForAjax
 
+Given(/^AGFS reform commenced on "([^"]*)"$/) do |agfs_reform_start_date|
+  allow(Settings).to receive(:agfs_fee_reform_release_date).and_return Date.parse(agfs_reform_start_date)
```

Comment:
Not sure about this, features specs, IMO should not be mocking internal logic of the application.

Reply:
simplest way too enable a feature test which we did not have before. without mocking we would not be able to test this feature. can you think of another way to simply get around the 3 month rule on warrant fees?

Reply:
💭 Configuring the `agfs_fee_reform_release_date` for the cucumber environment to be at least 3 months old.

For reference: https://github.com/cucumber/cucumber/wiki/Mocking-and-Stubbing-with-Cucumber

Reply:
Hmm, sounds like that would not be very declarative for a  given feature. that link suggests using stubs instead, which from reading on [RSpec mocks](https://relishapp.com/rspec/rspec-mocks/docs) the statement above is a stub. but `Settings.stub(:agfs_fee_reform_release_date).and_return(Date.parse(agfs_reform_start_date))` works just as well